### ISSUE TITLE
[R] R package roadmap

### ISSUE THREAD

**Issue by thirdwing** [apache/incubator-mxnet#3006]
This is for our roadmap for mxnet R package. Please feel free to add more. @tqchen @terrytangyuan @ziyeqinghan @hetong007 
- [x] Better method to set default ctx. #2392 
- [ ] Autoenconders in R. #1518 
- [x] Compatible with Rdata, integrate with caret. #362
- [x] RNN documents. @ziyeqinghan
- [x] More examples, including inception/googlenet/alexnet
- [x] Multi-GPU examples
- [x] Multi-Input Support #1269 #2266 
- [ ] Sparse Matrix Input Support #1524 #2902 
- [x] add code examples in the documentation
- [ ] Add a deep art example with mxnetR
- [x] Make the model object can be saved in RDS format

**Comment by yanyachen** [comment]
Multi-Input Support #1269 #2266 
Sparse Matrix Input Support #1524 #2902

**Comment by predict-r** [comment]
- [ ] add code examples in the documentation

**Comment by RogerBorras** [comment]
- [ ] Add a deep art example with mxnetR

**Comment by slavarazbash** [comment]
- Print methods for fitted model objects
- More detailed documentation. 
For example: 
>>list the possible values for out_activation in mx.mlp(). 
>>References to papers 
>>Some functions do not have examples. eg: mx.callback.log.train.metric()

**Comment by miguelgfierro** [comment]
- [ ] Solve issue with multi device in multi GPU #3633
- [ ] Add imagenet example #3832 
- [ ] Implement Speedometer

**Comment by RogerBorras** [comment]
- [ ] Add more RNN tutorials (for easy and advanced level)

**Comment by p2004r** [comment]
- [ ] Multi-Input Support

**Comment by topepo** [comment]
> Make the model object can be saved in RDS format

Any progress on this?

**Comment by thirdwing** [comment]
Really busy with the thesis. No time or energy for this now. @topepo

**Comment by ryangarner** [comment]
Why not just take RStudio's approach and use the [reticulate](https://github.com/rstudio/reticulate) package?

https://tensorflow.rstudio.com/
https://rstudio.github.io/keras/

I'm currently using reticulate to utilize the experimental Python RNN API in R.

I personally prefer working with dplyr for data munging and ggplot2 for data visualization instead of pandas and seaborn.

I also use [mlr](https://github.com/mlr-org/mlrpackage)  for hyperparameter tuning.
